### PR TITLE
Create League Modal

### PR DESCRIPTION
**Pull request by MichaelCduBois**
This PR creates the initial League model within the database. It is super simple, but we can add more features as time comes.

**This PR needs to be merged after [User Model](https://github.com/Leagueify/leagueify/pull/203)**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  id      Int     @id @default(autoincrement())
+  name    String  @db.VarChar(255)
+  sport   String   @db.VarChar(255)
+}
+
+model User {
+  id          Int       @id @default(autoincrement())
+  name        String    @db.VarChar(255)
+  email       String    @unique @db.VarChar(255)
+  phone       String    @db.VarChar(255)
```
Comment (kylekrny) [inline prisma/schema.prisma:20]: for varchars my idea

email: 60 
phone: 25 - allows for us to go international while helping to reduce spam and size. We can choose whether or not to save formatted or unformatted numbers into the db.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+model League {
+  id      Int     @id @default(autoincrement())
+  name    String  @db.VarChar(255)
+  sport   String   @db.VarChar(255)
+}
+
+model User {
+  id          Int       @id @default(autoincrement())
+  name        String    @db.VarChar(255)
+  email       String    @unique @db.VarChar(255)
```
[the remark this answers is not in the document]
Comment (MichaelCduBois) [reply prisma/schema.prisma:19]: You might be correct with that assumption, however we can ensure that we always format the provided string as lowercase. That would ensure we are unique.

File: prisma/schema.prisma (hunk 1 above)
Comment (MichaelCduBois) [reply prisma/schema.prisma:20]: I will be adjusting the phone and email length in the user model PR. Let's round emails to 64?

According to atdata.com regarding long email addresses:
>And if you want 95% of your web visitors to be able to see what they typed, you need to show at least 31 characters.

File: prisma/migrations/20230221031103_league/migration.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+-- CreateTable
+CREATE TABLE "League" (
+    "id" SERIAL NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "sport" VARCHAR(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (MichaelCduBois) [reply prisma/migrations/20230221031103_league/migration.sql:5]: I will be able to add that table quite easily. Do you see any need other than index and name for the sport to begin?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+model League {
+  id      Int     @id @default(autoincrement())
+  name    String  @db.VarChar(255)
+  sport   String   @db.VarChar(255)
+}
+
+model User {
+  id          Int       @id @default(autoincrement())
+  name        String    @db.VarChar(255)
```
[the remark this answers is not in the document]
Comment (MichaelCduBois) [reply prisma/schema.prisma:18]: We can definitely add a cap initially. I don't think we will have TOO many issues with 30, or 32, but if we do start getting errors, we should expand that. See this [discussion](https://ux.stackexchange.com/a/55532).

File: prisma/migrations/20230221031103_league/migration.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (kylekrny) [reply prisma/migrations/20230221031103_league/migration.sql]: Initially no. We can add defaults (team size, stats, game info, player roles, etc.) further down.

File: prisma/schema.prisma (hunk 2 above)
[the remark this answers is not in the document]
Comment (kylekrny) [reply prisma/schema.prisma]: That works for me.

File: prisma/schema.prisma (hunk 4 above)
[the remark this answers is not in the document]
Comment (kylekrny) [reply prisma/schema.prisma]: Happy to go up to 60. 60 is easy to style into a line. 255 is a small paragraph.

File: prisma/schema.prisma (hunk 1 above)
Comment (kylekrny) [reply prisma/schema.prisma]: works for me.

**Review by kylekrny — APPROVED** [review]

**Pull request merged**